### DISCUSSION TITLE
Understanding Invite only and Open batches

### DISCUSSION THREAD

**Discussion by Xitija** [Sunbird-Ed/Community#554] [Sunbird ED configuration discussion] [answered]
Hello All,

I am trying to enable open and invite-only batches for courses on Sunbird.

I have doubts regarding batch updation and users added to a batch in case of invite-only.

1. Once the batch is created of a particular type, either invite-only or open, we cannot modify its type like changing from invite-only to open. or vice versa. 

2. What type of users are displayed in the mentor and member sections?  

3. Which users can be added to the course batch? Do the users belong to the same organization?

4. What are the batch updation criteria, when can it be updated is the duration is till the batch starts, or is still ongoing?

![batchtk](https://github.com/Sunbird-Ed/Community/assets/65657373/f60d824a-b7ef-455a-a1ad-85e617cc105f)

Please guide me here.

cc: @coolbung @vinod303 @Renuka-S @komalm

**Comment by chitranshu-keshav** [comment]
@rajeevsathish @ashokreddy1208 can you please get back on this request

**Comment by rajeevsathish** [comment]
@Xitija 
1. Once the batch is created of a particular type, either invite-only or open, we cannot modify its type like changing from invite-only to open. or vice versa. Yes we cannot change the type once the batch is created
2. What type of users are displayed in the mentor and member sections?
if a course creator is trying to create a batch then in the mentors list only the users who as a COURSE_MENTOR role of the org to which the course creator is associated will only be available.
3.  Which users can be added to the course batch? Do the users belong to the same organization?
yes the user should belong to the same org in with the batch creator is and he should not be in the mentor list added already.
4. What are the batch updation criteria, when can it be updated is the duration is till the batch starts, or is still ongoing?
A batch start date can be updated any time till the batch is started
A Batch end date can be updated any time till the batch expires
A Batch ENROLMENT end date can be updated till the batch is not expired

cc: @chitranshu-keshav @ashokreddy1208 @vinukumar-vs

**Reply by Xitija** [reply]
@rajeevsathish Sir as per our discussion above 
My understanding is, we should not be able to edit the batch after its expiry but here edit is still enabled and I cannot create a new batch. Can you please help me understand this?
![batch-expired](https://github.com/Sunbird-Ed/Community/assets/65657373/19d0b668-fa22-4c43-b0da-d582a1410635)

**Reply by Xitija** [reply]
I have one more question regarding this, whether mentors and participants can be removed after adding them to the batch?

CC: @chitranshu-keshav @rajeevsathish @vinukumar-vs @coolbung

**Reply by Krishnaj20** [reply]
@ashokreddy1208 - can you respond to these queires? If needed, please get into a call with @Xitija to understand the query.

**Comment by rajeevsathish** [comment]
@Xitija you can change the batch by using the edit batch function and remove a user until he/she has not started consumption of the course post that you can remove but an entry in db will be there.
cc: @chitranshu-keshav @Krishnaj20 @ashokreddy1208 @vinukumar-vs

**Reply by Krishnaj20** [reply]
@Xitija - Can you also tag Vinod to the thread? He can contribute to this discussion.

**Reply by Xitija** [reply]
@rajeevsathish Sir
I checked if the user can be removed when I am editing a batch but, I am not able to find any button, that will remove the selected users - here for now I am just trying to remove the mentors I have added while I created the batch. 
![removing-selected-mentors-png-773×667-](https://github.com/Sunbird-Ed/Community/assets/65657373/aacf303e-bd70-4963-a3b6-ea4030323dc0)


cc: @vinod303

**Reply by Xitija** [reply]
I need to understand that the response of batch details for a batch that is already created
contains a status whose value is 1. What does that status indicate?
Following is the batch detail response 
`{
    "identifier": "0138341243733196803",
    "createdFor": [
        "0130701891041689600"
    ],
    "endDate": "2023-07-08",
    "description": "",
    "updatedDate": null,
    "batchId": "0138341243733196803",
    "cert_templates": null,
    "createdDate": "2023-07-07 18:31:48:529+0530",
    "createdBy": "5d757783-a86a-40cd-a814-1b6a16d37cb6",
    "mentors": [
        "49e8b096-9d70-4cfa-b05a-3de649447985",
        "fe5b7136-3232-4b7a-9dab-8a3f2cf5c7ef"
    ],
    "name": "Batch A",
    "id": "0138341243733196803",
    "enrollmentType": "open",
    "courseId": "do_1138233452695797761116",
    "enrollmentEndDate": "2023-07-08",
    "collectionId": "do_1138233452695797761116",
    "startDate": "2023-07-07",
    "status": 1
}
`

@maheshkumargangula

**Reply by Krishnaj20** [reply]
@amiableanil - Can you check and respond to this thread asap?

@rajeevsathish - this thread isn't closed. Pls tag the right people to close the thread.

**Reply by AmiableAnil** [reply]
@Xitija please refer in microsite here https://lern.sunbird.org/learn/product-and-developer-guide/batch-service/features-core-capabilities#how-is-the-batch-status-updated

**Reply by vinod303** [reply, marked as answer]
@Xitija lets have a call to clarify the points mentioned in this thread
As mentioned in the doc, the Batch status = 1 indicate that it is a live batch. ie. the users can still consume the course and their progress etc.. will get updated.

**Reply by Krishnaj20** [reply]
@vinod303 - I'm hoping the queries are resolved via your call with @Xitija .

**Reply by vinod303** [reply]
@Krishnaj20  clarified  @Xitija 's questions over a call.  Thanks to @rajeevsathish also for his inputs for the same.
